TestDriveRemotesEndToEnd data race in gowebdav

https://github.com/tailscale/tailscale/actions/runs/28398124627/job/84142149004?pr=20276

```
?	tailscale.com/ipn/ipnauth [skipped/no tests] 
?	tailscale.com/ipn/ipnext [skipped/no tests] 
ok	tailscale.com/ipn	1.201s
ok	tailscale.com/ipn/auditlog	1.085s
=== RUN   TestDriveRemotesEndToEnd
==================
WARNING: DATA RACE
Read at 0x00c000556870 by goroutine 248:
  strings.(*Reader).Read()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/strings/reader.go:40 +0x55
  io.(*nopCloserWriterTo).Read()
      <autogenerated>:1 +0x5b
  net/http.(*readTrackingBody).Read()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:775 +0x76
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:429 +0x270
  net/http/internal.(*chunkedWriter).Write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/internal/chunked.go:248 +0xf1
  net/http/internal.(*chunkedWriter).Write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/internal/chunked.go:245 +0xb1
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:431 +0x2c1
  io.(*multiReader).writeToWithBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:54 +0x153
  net/http/internal.(*chunkedWriter).Write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/internal/chunked.go:245 +0xb1
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:431 +0x2c1
  io.(*multiReader).writeToWithBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:54 +0x153
  io.(*multiReader).WriteTo()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:45 +0x59
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:411 +0xd3
  io.CopyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:402 +0x8f
  net/http.(*transferWriter).doBodyCopy()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transfer.go:417 +0x13a
  net/http.(*transferWriter).writeBody()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transfer.go:361 +0xef3
  net/http.(*Request).write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/request.go:762 +0x1376
  net/http.(*persistConn).writeLoop()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2661 +0x35d
  net/http.(*Transport).dialConn.gowrap3()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2001 +0x2e

Previous write at 0x00c000556870 by goroutine 252:
  strings.(*Reader).Read()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/strings/reader.go:45 +0x105
  io.(*nopCloserWriterTo).Read()
      <autogenerated>:1 +0x5b
  net/http.(*readTrackingBody).Read()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:775 +0x76
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:429 +0x270
  io.(*multiReader).writeToWithBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:54 +0x153
  net/http/internal.(*chunkedWriter).Write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/internal/chunked.go:245 +0xb1
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:431 +0x2c1
  io.(*multiReader).writeToWithBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:54 +0x153
  io.(*multiReader).WriteTo()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/multi.go:45 +0x59
  io.copyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:411 +0xd3
  io.CopyBuffer()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/io/io.go:402 +0x8f
  net/http.(*transferWriter).doBodyCopy()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transfer.go:417 +0x13a
  net/http.(*transferWriter).writeBody()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transfer.go:361 +0xef3
  net/http.(*Request).write()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/request.go:762 +0x1376
  net/http.(*persistConn).writeLoop()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2661 +0x35d
  net/http.(*Transport).dialConn.gowrap3()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2001 +0x2e

Goroutine 248 (running) created at:
  net/http.(*Transport).dialConn()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2001 +0x303c
  net/http.(*Transport).dialConnFor()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:1654 +0x124
  net/http.(*Transport).startDialConnForLocked.func1()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:1635 +0x3c

Goroutine 252 (running) created at:
  net/http.(*Transport).dialConn()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:2001 +0x303c
  net/http.(*Transport).dialConnFor()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:1654 +0x124
  net/http.(*Transport).startDialConnForLocked.func1()
      /home/runner/.cache/tsgo/c803676bcc7f2b195b167a53d49d728045cd9b36/src/net/http/transport.go:1635 +0x3c
==================
    testing.go:1712: race detected during execution of test
2026/06/29 19:47:40 serve: invalid argument
--- FAIL: TestDriveRemotesEndToEnd (0.04s)
?	tailscale.com/ipn/ipnlocal/ipnlocaltest [skipped/no tests] 
ok	tailscale.com/ipn/ipnlocal/netmapcache	1.090s
FAIL
=== RUN   TestIsNotableNotify
--- PASS: TestIsNotableNotify (0.00s)
=== RUN   TestWriteNetmapDoesNotMutateOriginal
--- PASS: TestWriteNetmapDoesNotMutateOriginal (0.09s)
=== RUN   TestDNSConfigForNetmap
=== RUN   TestDNSConfigForNetmap/empty
--- PASS: TestDNSConfigForNetmap/empty (0.00s)
=== RUN   TestDNSConfigForNetmap/self_name_and_peers
```

--- a/drive/driveimpl/drive_test.go
+++ b/drive/driveimpl/drive_test.go
@@ -522,7 +522,7 @@ func newSystem(t *testing.T) *system {
 		}
 	}()
 
-	client := gowebdav.NewAuthClient(fmt.Sprintf("http://%s", ln.Addr()), &noopAuthorizer{})
+	client := gowebdav.NewClient(fmt.Sprintf("http://%s", ln.Addr()), "", "")
 	client.SetTransport(&http.Transport{DisableKeepAlives: true})
 	s := &system{
 		t:     t,
@@ -770,36 +770,6 @@ func fileInfoToStatic(fi fs.FileInfo, fixupMode bool) fs.FileInfo {
 
 func pathTo(remote, share, name string) string {
 	return path.Join(domain, remote, share, name)
-}
-
-// noopAuthorizer implements gowebdav.Authorizer. It does no actual
-// authorizing. We use it in place of gowebdav's built-in authorizer in order
-// to avoid a race condition in that authorizer.
-type noopAuthorizer struct{}
-
-func (a *noopAuthorizer) NewAuthenticator(body io.Reader) (gowebdav.Authenticator, io.Reader) {
-	return &noopAuthenticator{}, nil
-}
-
-func (a *noopAuthorizer) AddAuthenticator(key string, fn gowebdav.AuthFactory) {
-}
-
-type noopAuthenticator struct{}
-
-func (a *noopAuthenticator) Authorize(c *http.Client, rq *http.Request, path string) error {
-	return nil
-}
-
-func (a *noopAuthenticator) Verify(c *http.Client, rs *http.Response, path string) (redo bool, err error) {
-	return false, nil
-}
-
-func (a *noopAuthenticator) Clone() gowebdav.Authenticator {
-	return &noopAuthenticator{}
-}
-
-func (a *noopAuthenticator) Close() error {
-	return nil
 }
 
 const lockBody = `<?xml version="1.0" encoding="utf-8" ?>

--- a/flake.nix
+++ b/flake.nix
@@ -169,4 +169,4 @@
     });
   };
 }
-# nix-direnv cache busting line: sha256-PnzuJF0Rz2gG7Ki9KDOlHi2dhGysqLZTKjIaGWGNPjk=
+# nix-direnv cache busting line: sha256-7synjuzpYZqDwIGVLqN9gmwDFDmWUS7a7rRLUnJIPK4=

--- a/flakehashes.json
+++ b/flakehashes.json
@@ -4,7 +4,7 @@
     "sri": "sha256-TwIaPmGVHO9ZwdaO/jDStgWGQtKa4smS3er1i5aTNTw="
   },
   "vendor": {
-    "goModSum": "sha256-oT5vCkW7a28PnVTeD8zfYbPe54uGvvfpiicrZFx8p+Y=",
-    "sri": "sha256-PnzuJF0Rz2gG7Ki9KDOlHi2dhGysqLZTKjIaGWGNPjk="
+    "goModSum": "sha256-sUoV/3avVCbUmIrhdmYeQ47noNXWxYAxEEQ5wpsQ7O4=",
+    "sri": "sha256-7synjuzpYZqDwIGVLqN9gmwDFDmWUS7a7rRLUnJIPK4="
   }
 }

--- a/go.mod
+++ b/go.mod
@@ -96,7 +96,7 @@ require (
 	github.com/robert-nix/ansihtml v1.0.1
 	github.com/safchain/ethtool v0.3.0
 	github.com/skip2/go-qrcode v0.0.0-20200617195104-da1b6568686e
-	github.com/studio-b12/gowebdav v0.9.0
+	github.com/studio-b12/gowebdav v0.13.0
 	github.com/tailscale/certstore v0.1.1-0.20260409135935-3638fb84b77d
 	github.com/tailscale/depaware v0.0.0-20251001183927-9c2ad255ef3f
 	github.com/tailscale/gliderssh v0.3.4-0.20260603235856-90d7dcbcbf38

--- a/shell.nix
+++ b/shell.nix
@@ -16,4 +16,4 @@
 ) {
   src =  ./.;
 }).shellNix
-# nix-direnv cache busting line: sha256-PnzuJF0Rz2gG7Ki9KDOlHi2dhGysqLZTKjIaGWGNPjk=
+# nix-direnv cache busting line: sha256-7synjuzpYZqDwIGVLqN9gmwDFDmWUS7a7rRLUnJIPK4=

